src/main/java/com/telekom/m2m/cot/restsdk/realtime/Notification.java

```
@@ -13,6 +14,32 @@ public Notification(JsonObject data) {
         this.data = data;
     }
 
+    /**
+     * get the realtime action out of the json data
+     * @return enum value for realtime action
+     */
+    public Enum<RealtimeAction> getRealtimeAction() {
+        try {
+            String realtimeAction = data.get("realtimeAction").getAsString();
+            return RealtimeAction.valueOf(realtimeAction);
+        } catch (NullPointerException e) {
+            throw new CotSdkException("No member realtimeAction in Json Object");
+        }
+
+    }
+
+    /**
+     *
+     * @return the data part of the json data
+     */
+    public String getDataPart() {
+        try {
+            return data.get("data").getAsString();
+        } catch (NullPointerException e) {
+            throw new CotSdkException("No member data in Json Object");
```

The original exception is lost here.
The exception should always be passed as previous exception, otherwise the user does not have the chance to find the root cause of this error.